crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/CrownCourtEligibilityService.java

```diff
@@ -61,10 +58,15 @@ public boolean isEligibilityCheckRequired(MeansAssessmentRequestDTO assessmentRe
                     if (previousAssessment != null) {
                         return !hasDisqualifyingResult(previousAssessment);
                     }
+                    return true;
                 }
             }
         }
-        return true;
+        Stream<Assessment> previousAssessments = getPreviousAssessments(repOrder);
+        boolean hasDisqualifyingResult = previousAssessments
+                .anyMatch(this::hasDisqualifyingResult);
```

Comment:
The logic hear can be simplified slightly by removing the logical negation (!) and using the `noneMatch` method.
e.g.

```
        boolean hasDisqualifyingResult = previousAssessments
                .noneMatch(this::hasDisqualifyingResult);

        return hasDisqualifyingResult;
```

Reply:
Good suggestion, that greatly simplifies the logic.